pkg/operator/ceph/csi/spec.go

```diff
@@ -734,6 +735,10 @@ func (r *ReconcileCSI) validateCSIVersion(ownerInfo *k8sutil.OwnerInfo) (*CephCS
 	}
 	logger.Infof("Detected ceph CSI image version: %q", version)
 
+	// Set the CSI version on a package variable so that each cluster reporting the
+	// telemetry can report the same CSI version
+	telemetry.CSIVersion = version.String()
```

Comment:
reporting sidecar versions are also useful.

>rook/usage/storage-class/count/total
rook/usage/storage-class/count/rbd
rook/usage/storage-class/count/cephfs
rook/usage/storage-class/count/nfs
rook/usage/storage-class/count/bucket

This will also take rados namespace and subvolumegroup into consideration?

The above metrics are per ceph cluster or consolidated output of all clusters.

>rook/node/count/with-csi-rbd-plugin - Number of nodes with CSI RBD plugin pods
rook/node/count/with-csi-cephfs-plugin - Number of nodes with CSI CephFS plugin pods
rook/node/count/with-csi-nfs-plugin - Number of nodes with CSI NFS plugin pods

Can we add the holder pod to this list also?

Reply:
Do we need to add details about which csi driver is enabled? as we have the option to disable csi driver also?

Reply:
All of those are good considerations. Could you open a new issue for these suggestions? We will need to add all metrics first to the [telemetry design doc](https://github.com/rook/rook/blob/master/design/ceph/ceph-telemetry.md) before we can get implement. Thanks!

Reply:
@travisn https://github.com/rook/rook/issues/10402 created this one.